agent/api/task.go

```diff
@@ -847,6 +867,24 @@ func (task *Task) GetTaskENI() *ENI {
 	return task.ENI
 }
 
+// ShouldWaitForExecutionCredentials check if there are container waiting for the
+// credentials to progress, for example: to pull image from ECR
+func (task *Task) ShouldWaitForExecutionCredentials() bool {
```

Comment:
I don't think the task should ever wait, but the containers might.  This doesn't seem appropriate at the task level.

Reply:
this is no longer needed, I will remove this.